polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -309,41 +339,59 @@ class ScalaCompiler private (
       case DefDef(mods, name, _, _, _, _) if mods.isPublic => name
     }.distinct
 
+    lazy val wrappedImports = copyAndReset(inheritedImports.externalImports) ++ copyAndReset(inheritedImports.localImports)
+
     // The code all wrapped up in a class definition, with constructor arguments for the given prior cells and inputs
     private lazy val wrappedClass: ClassDef = {
-      val priorCellParamList = copyAndReset(priorCellInputs)
-      val nonImplicitParamList = copyAndReset(nonImplicitInputs)
-      val implicitParamList = copyAndReset(implicitInputs)
-      q"""
-        final class $assignedTypeName(..$priorCellParamList)(..$nonImplicitParamList)(..$implicitParamList) extends scala.Serializable {
-          ..${priorCellImports}
-          ..${copyAndReset(inheritedImports.externalImports)}
-          ..${copyAndReset(inheritedImports.localImports)}
-          ..${compiledCode}
-        }
-      """
+      val transparentBefore = beforePos.makeTransparent
+
+      def toParam(param: ValDef) = atPos(transparentBefore)(param.copy(mods = param.mods | Flag.PARAMACCESSOR))
+      val priorCellParamList = copyAndReset(priorCellInputs).map(toParam)
+      val nonImplicitParamList = copyAndReset(nonImplicitInputs).map(toParam)
+      val implicitParamList = copyAndReset(implicitInputs).map(toParam)
+
+      // have to do this manually rather than with a quasiquote; latter messes up positions
```

Comment:
Here I had to move away from the quasiquote, because it blows up if anything is amiss with positions. I guess quasiquotes assign positions with no source file, so it screws things up down the road.

Note that the quasiquote `ClassDef` constructor does a bunch of stuff that `mkClassDef` doesn't do (for some reason) which turns out to be completely necessary. If we don't add the `PARAMACCESSOR` flag to the constructor params, then they don't properly become constructor parameters (I guess?) and there will be an NPE at runtime when we instantiate the class.